### PR TITLE
update README w deepspeed info

### PR DESCRIPTION
**Pull request by winglian**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -732,6 +735,21 @@ wandb_run_id:
 wandb_log_model:
 ```
 
+### Training with Deepspeed
```
Comment (NanoCode012) [inline README.md:731]: We do have a short explanation on using deep speed on line 723. perhaps these two section can be integrated or the earlier removed.

File: README.md [hunk 2]
```diff
@@ -88,6 +89,8 @@ cd axolotl
 pip3 install packaging
 pip3 install -e .[flash-attn]
 pip3 install -U git+https://github.com/huggingface/peft.git
+# optionally install deepspeed
```
Comment (irbull) [inline README.md]: See https://github.com/OpenAccess-AI-Collective/axolotl/issues/598, `deepspeed` doesn't appear to be optional anymore. If it's not installed, Axolotl won't run.

**Comment by NanoCode012** [comment]
Should we just put deepspeed into requirements.txt?

File: setup.py [hunk 3]
```diff
@@ -13,7 +13,12 @@ def parse_requirements():
                 # Handle custom index URLs
                 _, url = line.split()
                 _dependency_links.append(url)
-            elif "flash-attn" not in line and line and line[0] != "#":
+            elif (
+                "flash-attn" not in line
```
Comment (NanoCode012) [inline setup.py:17]: Is there a reason to ignore deepspeed?

File: README.md [hunk 4]
```diff
@@ -732,6 +728,27 @@ wandb_run_id:
...
+
+Deepspeed is an optimization suite for multi-gpu systems allowing you to train much larger models than you
+might typically be able to fit into your GPU's VRAM. More information about the various optimization types
+for deepspeed is available at https://huggingface.co/docs/accelerate/main/en/usage_guides/deepspeed#what-is-integrated
+
+We provide several default deepspeed JSON configurations for ZeRO stage 1, 2, and 3.
+
+```shell
+# install deepspeed if you haven't already
+pip3 install deepspeed
```
Comment (NanoCode012) [inline README.md]: This pip install is likely redundant if it's included in main instructions.

File: setup.py (hunk 3 above)
Comment (winglian) [reply setup.py:17]: deepspeed doesn't consistently install on all environments cleanly, esp WSL

Comment (NanoCode012) [reply setup.py:17]: Ah I see

**Pull request merged**